src/main/scala/nl.knaw.dans.easy.ccw/EasyCollectCurationWorkApp.scala

```diff
@@ -15,17 +15,71 @@
  */
 package nl.knaw.dans.easy.ccw
 
-import scala.util.{ Success, Try }
+import better.files.File
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
 
-class EasyCollectCurationWorkApp(configuration: Configuration)  {
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+import scala.util.Try
 
-  def run(): Try[Unit] = {
+class EasyCollectCurationWorkApp(commonCurationArea: File, managerCurationDirString: String, datamanagerProperties: PropertiesConfiguration) extends DebugEnhancedLogging {
 
+  private def isCurated(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getBoolean("curation.performed")
+  }
 
+  private def requiredPropertiesPresent(datamanager: DatamanagerId, deposit: File, depositProperties: PropertiesConfiguration, keys: List[String]): Boolean = {
+    var requiredPropertiesPresent = true
+    for (key <- keys
+         if depositProperties.getString(key, "").isEmpty) {
+      logger.error(s"Deposit ${ deposit.name }, curated by $datamanager, has no value for property '$key'")
+      requiredPropertiesPresent = false
+    }
+    requiredPropertiesPresent
+  }
 
+  private def correctState(datamanager: DatamanagerId, deposit: File, state: String): Boolean = {
+    if (state != "SUBMITTED" && state != "REJECTED") {
+      logger.error(s"Deposit ${ deposit.name }, curated by $datamanager, is curated, but is in state $state")
+      false
+    }
+    else
+      true
+  }
 
-    ???
+  private def correctProperties(datamanager: DatamanagerId, deposit: File, depositProperties: PropertiesConfiguration, state: String): Boolean = {
+    val propertiesToCheck = List("state.description", "curation.datamanager.userId", "curation.datamanager.email")
+    val isCorrectState = correctState(datamanager, deposit, state)
+    requiredPropertiesPresent(datamanager, deposit, depositProperties, propertiesToCheck) && isCorrectState
   }
 
+  private def collectDeposit(datamanager: DatamanagerId, deposit: File): Unit = {
+    val depositProperties = new PropertiesConfiguration(deposit / "deposit.properties" toJava)
+    val state = depositProperties.getString("state.label", "")
 
+    if (isCurated(depositProperties) && correctProperties(datamanager, deposit, depositProperties, state)) {
+      deposit moveTo commonCurationArea / deposit.name
+      logger.info(s"Deposit ${ deposit.name } ($state), curated by $datamanager, has been moved to common curation area")
+    }
+  }
+
+  private def collectDatamanagersCuratedDeposits(datamanager: DatamanagerId, dir: File): Unit = {
+    dir.list.toList.foreach(deposit => collectDeposit(datamanager, deposit))
```

Comment:
don't need the extra overhead of `toList` conversion here: `dir.list` gives an `Iterator[File]`, which is good enough for this `foreach`